### PR TITLE
Allow BasicAuthProvider to use hashed passwords

### PR DESCRIPTION
**Pull request by pbaumard**
Adding tests for this PR does not seem straightforward and I have not much time for it.

### CODE REVIEW

File: FROST-Server.Auth.Basic/src/main/java/de/fraunhofer/iosb/ilt/frostserver/auth/basic/BasicAuthProvider.java [hunk 1]
```diff
@@ -38,6 +38,9 @@ public class BasicAuthProvider implements AuthProvider, LiquibaseUser, ConfigDef
     @DefaultValueBoolean(false)
     public static final String TAG_AUTO_UPDATE_DATABASE = "autoUpdateDatabase";
 
+    @DefaultValueBoolean(true)
+    public static final String TAG_PLAIN_TEST_PASSWORD = "plainTextPassword";
```
Comment (hylkevds) [inline FROST-Server.Auth.Basic/src/main/java/de/fraunhofer/iosb/ilt/frostserver/auth/basic/BasicAuthProvider.java]: Typo in the name of the constant? (TEST -> TEXT)

**Comment by hylkevds** [comment]
Thanks for the PR!
I'll look into adding the tests for it. Should be relatively simple by copying the current tests for the BasicAuth filter.

It should probably include a little text in the documentation about how to set the passwords in the database.

File: FROST-Server.Auth.Basic/src/main/java/de/fraunhofer/iosb/ilt/frostserver/auth/basic/BasicAuthProvider.java (hunk 1 above)
Comment (pbaumard) [reply FROST-Server.Auth.Basic/src/main/java/de/fraunhofer/iosb/ilt/frostserver/auth/basic/BasicAuthProvider.java]: I fixed the typo.

**Comment by pbaumard** [comment]
> It should probably include a little text in the documentation about how to set the passwords in the database.

I added a SQL sample.

**Comment by hylkevds** [comment]
I've added the tests in a separate commit, and updated the Liquibase scripts to generate default users with hashed password if the option is turned on.

**Pull request merged**